sources/views/contact-form.js

```diff
@@ -0,0 +1,247 @@
+									validate: webix.rules.isNotEmpty,
+									invalidMessage: "Can not be empty"
+								},
+								{
+									view: "text",
+									label: "Address",
+									name: "Address",
+									validate: webix.rules.isNotEmpty,
+									invalidMessage: "Can not be empty"
+								}
+							]
+						},
+						{
+							margin: 20,
+							rows: [
+								{
+									view: "text",
+									label: "Email",
+									name: "Email",
+									validate: webix.rules.isNotEmpty,
+									invalidMessage: "Can not be empty"
+								},
+								{
+									view: "text",
+									label: "Skype",
+									name: "Skype",
+									validate: webix.rules.isNotEmpty,
+									invalidMessage: "Can not be empty"
+								},
+								{
+									view: "text",
+									label: "Phone",
+									name: "Phone",
+									validate: webix.rules.isNotEmpty,
+									invalidMessage: "Can not be empty"
+								},
+								{
+									view: "datepicker",
+									label: "Birthday",
+									name: "Birthday",
+									format: webix.i18n.longDateFormatStr,
+									validate: webix.rules.isNotEmpty,
+									invalidMessage: "Can not be empty"
+								},
+								{
+									cols: [
+										{
+											type: "clean",
+											localId: "contactPhoto",
+											name: "Photo",
+											height: 200,
+											gravity: 2,
+											template: contact => `<div class="user-info-photo">
+											<img src=${contact.Photo || noPhoto} alt="User photo" /></div>`
+										},
+										{
+											rows: [
+												{},
+												{
+													view: "uploader",
+													localId: "photoUploader",
+													label: "Change photo",
+													accept: "image/png, image/jpg, image/jpeg",
+													autosend: false,
+													css: "button--style"
+												},
+												{
+													view: "button",
+													label: "Delete photo",
+													css: "button--style",
+													click: () => this.deletePhoto()
+												}
+											]
+										}
+									]
+								}
+							]
+						}
+					]
+				},
+				{},
+				{
+					cols: [
+						{view: "button", label: "Cancel", css: "button--style", click: () => this.closeForm()},
+						{
+							view: "button",
+							label: "",
+							css: "button--style",
+							localId: "actionsButton",
+							click: () => this.addEditContact()
+						}
+					]
+				}
+			]
+		};
+		return form;
+	}
+
+	init() {
+		this.form = this.$$("contactForm");
+		this.contactPhoto = this.$$("contactPhoto");
+		const fileReader = new FileReader();
+
+		this.$$("photoUploader").attachEvent("onBeforeFileAdd", (data) => {
+			fileReader.readAsDataURL(data.file);
+			fileReader.onload = () => this.contactPhoto.setValues({Photo: fileReader.result});
+		});
+	}
+
+	urlChange() {
+		const state = this.getParam("status");
+		if (state === "edit") {
+			const id = this.getParam("id", true);
+			this.showContactForm(state, id);
+		}
+		else {
+			this.showContactForm(state);
```

Comment:
Would be better to clear the form on clicking Add item. 
To see the issue, 
- open editing form for any contact
- click on "Add contact" button 
- mode in URL will be changed, but id and actual data will remain (as a result, form will edit the existing item, as it have `id`)

Reply:
Fixed.

Reply:
`this.form.clearValidation();` (`clear` as well, as it is natural to get rid of old data) can be used as the 1st thing before `if` statement at
https://github.com/KarachunVladislav01/jet-test-app/blob/fe4587f81ec4b65dff6a5c7be5bff8b13ec4240b/sources/views/contact-form.js#L194

Otherwise, validation marks will remain if you switch between contacts:
- open editing form for any contact
- delete the name, try to save it => validation message appears
- switch to another contact => validation remains

Reply:
Fixed.